runtime/zeitgeist/src/integration_tests/xcm/tests/transfers.rs

```diff
@@ -0,0 +1,336 @@
+// Zeitgeist is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at
+// your option) any later version.
+//
+// Zeitgeist is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Zeitgeist. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{
+    integration_tests::xcm::{
+        setup::{
+            ksm, register_foreign_parent, register_foreign_ztg, sibling_account, zeitgeist_account,
+            ztg, ALICE, BOB, FOREIGN_PARENT_ID, FOREIGN_ZTG_ID, PARA_ID_SIBLING,
+        },
+        test_net::{KusamaNet, Sibling, TestNet, Zeitgeist},
+    },
+    xcm_config::{
+        asset_registry::{CustomMetadata, XcmMetadata},
+        config::zeitgeist,
+        fees::default_per_second,
+    },
+    AssetRegistry, Balance, Balances, CurrencyId, Origin, Tokens, XTokens,
+    ZeitgeistTreasuryAccount,
+};
+
+use frame_support::assert_ok;
+use orml_traits::MultiCurrency;
+use xcm::latest::{Junction, Junction::*, Junctions::*, MultiLocation, NetworkId};
+use xcm_emulator::TestExt;
+use zeitgeist_primitives::constants::BalanceFractionalDecimals;
+
+#[test]
+fn transfer_ztg_to_sibling() {
+    TestNet::reset();
+
+    let alice_initial_balance = ztg(10);
+    let transfer_amount = ztg(5);
+    let mut treasury_initial_balance = 0;
+
+    Sibling::execute_with(|| {
+        treasury_initial_balance =
+            Tokens::free_balance(FOREIGN_ZTG_ID, &ZeitgeistTreasuryAccount::get());
+        assert_eq!(Tokens::free_balance(FOREIGN_ZTG_ID, &BOB.into()), 0);
+        register_foreign_ztg(None);
+    });
+
+    Zeitgeist::execute_with(|| {
+        assert_eq!(Balances::free_balance(&ALICE.into()), alice_initial_balance);
+        assert_eq!(Balances::free_balance(&sibling_account()), 0);
+        assert_ok!(XTokens::transfer(
+            Origin::signed(ALICE.into()),
+            CurrencyId::Ztg,
+            transfer_amount,
+            Box::new(
+                MultiLocation::new(
+                    1,
+                    X2(
+                        Parachain(PARA_ID_SIBLING),
+                        Junction::AccountId32 { network: NetworkId::Any, id: BOB.into() }
+                    )
+                )
+                .into()
+            ),
+            4_000_000_000,
+        ));
+
+        // Confirm that Alice's balance is initial_balance - amount_transferred
+        assert_eq!(Balances::free_balance(&ALICE.into()), alice_initial_balance - transfer_amount);
+
+        // Verify that the amount transferred is now part of the sibling account here
+        assert_eq!(Balances::free_balance(&sibling_account()), transfer_amount);
```

Comment:
Yes, in the case of Zeitgeist that is correct.
Most of the parachains select them selves as reserves for their native currency. When funds are transferred, they are actually just moved in an account that is owned by the parachain and is unique to each parachain. The parachain account holds the reserve of the funds that are transferred to another chain. So in contract to teleports, where assets are really minted and burned, reserve asset transfers just put the reserved assets to a trusted location. If the chain acts as its own reserve, it's called `SelfReserve`. When this concept was presented initially, it included a third trusted party, like the parent chain. The ZTG would then be stored in a sovereign account on Kusama for example. However, if everyone does this, it can lead to heavy centralization.

Storing the reserves in multiple sovereign accounts can introduce some problems. Assume we have no sovereign accounts on the Zeitgeist chain yet. ZTG is moved to parachain B and consequently stored in the sovereign account of parachain B. Parachain B now moves ZTG to parachain C. When parachain C moves ZTG to Zeitgeist, it can't do so because it's sovereign account is empty.

Reply:
So with reserve transfers the sovereign account of parachain B would tell Zeitgeist about the transfer to C, so that the amount is transferred from B to C? Then C could sent ZTG to Zeitgeist again.

Reply:
I don't think that this is what happens currently, there is no additional XCM afaik. In the scenario above, B would act as a ZTG reserve for C.